common/djangoapps/custom_settings/views.py

```diff
@@ -0,0 +1,60 @@
+import json
+import logging
+
+import django.utils
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseBadRequest, HttpResponseNotAllowed
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_http_methods
+from opaque_keys.edx.keys import CourseKey
+
+from cms.djangoapps.contentstore.views.course import get_course_and_check_access
+from edxmako.shortcuts import render_to_response
+from util.json_request import JsonResponse, expect_json
+from util.views import require_global_staff
+from xmodule.modulestore.django import modulestore
+from .models import CustomSettings
+
+log = logging.getLogger(__name__)
+
+
+@require_global_staff
+@ensure_csrf_cookie
+@require_http_methods(("GET", "POST", "PUT"))
+@expect_json
+def course_custom_settings(request, course_key_string):
+    """
+    Course custom settings configuration
+    GET
+        html: get the page
+    PUT, POST
+        json: update the Course's custom settings.
+    """
+
+    course_key = CourseKey.from_string(course_key_string)
+
+    with modulestore().bulk_operations(course_key):
+        course_module = get_course_and_check_access(course_key, request.user)
+        try:
+            settings = CustomSettings.objects.get(id=course_key)
+        except ObjectDoesNotExist as exc:
+            return HttpResponseBadRequest(django.utils.html.escape(exc.message), content_type="text/plain")
+
+        if 'text/html' in request.META.get('HTTP_ACCEPT', '') and request.method == 'GET':
+            return render_to_response('custom_settings.html', {
+                'context_course': course_module,
+                'custom_dict': {'is_featured': settings.is_featured},
+                'custom_settings_url': reverse('custom_settings', kwargs={'course_key_string': unicode(course_key)}),
+            })
+
+        elif 'application/json' in request.META.get('HTTP_ACCEPT', '') and request.method in ['POST', 'PUT']:
+            body = json.loads(request.body)
+            is_featured = body.get('is_featured')
+            if is_featured in [True, False]:
```

Comment:
`is_featured ` is a boolean parameter. It should either be true or false. If you are getting something else in this param you need to fix the client instead of adding statements in the code which does not make any sense.

Reply:
Followed the pattern as it was in **advanced_settings_handler** view. Let me investigate this further.

Reply:
Check removed from backend applied on frontend.